### ISSUE TITLE
Modified sources after building

### ISSUE THREAD

**Issue by fsteggink** [heron-mc/heron-mc#461]
Hi, after building Heron, I'm left with some modified sources. This is the output of git status:
```
On branch master
Your branch is up-to-date with 'origin/master'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)
  (commit or discard the untracked or modified content in submodules)

        modified:   docsrc/examples.rst
        modified:   heron/ux/gxp/git (modified content)

no changes added to commit (use "git add" and/or "git commit -a")
```

In heron\ux\gxp\git the output of git status is:
```
HEAD detached at fd6627e
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        deleted:    externals/DimensionManager

no changes added to commit (use "git add" and/or "git commit -a")
```
(The commit has of gxp is the correct one)

A proper solution in the build process is very much appreciated.

See also pull request #460 in which I added a few .gitignore files to hide build output. However, those are hiding new files and dirs which are not part of the original source.

**Comment by justb4** [comment]
Possibly we can close this now PR #460 has been merged?

**Comment by fsteggink** [comment]
Unfortunately not. The gitignore files are specifically meant to hide _new files_ which are created by building Heron. This issue refers to _existing_ files.
Docsrc/examples.rst is one of the files which is updated by invoking the build. It is not updated manually by myself.

**Comment by justb4** [comment]
Yes, it's a bit tricky, like the examples.rst is a template that is expanded during build, eventually rendering http://heron-mc.org/examples.html. This provides minimal maintenance for the set of examples.  In ``heron/ux/gxp/git``a new gxp .js is built. 

I understand these files get in the way. I need to dive into the build process to see how we can keep the source tree clean...
